### PR TITLE
asserts: introduce new parseHeaders

### PR DESCRIPTION
**Pull request by pedronis**
This introduces the new parseHeaders returning map[string]interface{} and capable of accepting:

- the new "  - " prefixed and indented lists
- the adapted 4-space prefixed multiline text blocks
- nesting/mixing of those using the last anchor as the base for further indenting
 
The code doesn't uses it yet, just renames the previous incarnation to parseHeadersNaive for now.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #1581

File: asserts/headers.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		if err != nil {
+			return nil, err
+		}
+
+		headers[name] = value
+	}
+	return headers, nil
+}
+
+func nestingPrefix(nesting int, prefix string) string {
```
Comment (emgee) [inline asserts/headers.go]: tiny point, but maybe rename the "nesting" parameter to "level" or "depth"?

File: asserts/headers.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	return strings.Repeat("    ", nesting) + prefix
+}
+
+func parseEntry(consumedByIntro int, i int, lines []string, nesting int) (value interface{}, after int, err error) {
```
Comment (emgee) [inline asserts/headers.go]: could maybe rename i to start/first and after to end/last to make the relationship more obvious?

File: asserts/headers.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+			// list
+			return parseList(i, lines, nesting)
+		}
+
+		return parseMultilineText(i, lines, nesting)
+	}
+
+	// simple one-line value
+	if entry[consumedByIntro] != ' ' {
+		return nil, -1, fmt.Errorf("header entry should have a space or newline (multiline) before value: %q", entry)
```
Comment (emgee) [inline asserts/headers.go]: might be nice to add the number of the failing line to this error, but it only really matters once we have nested maps so is not too important right now.

File: asserts/headers.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	for j < len(lines) {
+		iline := lines[j]
+		if !strings.HasPrefix(iline, prefix) {
+			break
+		}
+		size += len(iline) - len(prefix) + 1
+		j++
+	}
+	if j == i {
+		return nil, -1, fmt.Errorf("expected indentation %q after multiline text introduction: %q", prefix, lines[i-1])
```
Comment (emgee) [inline asserts/headers.go]: include line number?

Comment (emgee) [inline asserts/headers.go]: and is this tested?

**Comment by chipaca** [comment]
:+1:

**Comment by emgee** [comment]
+1, and thanks for the name changes

**Pull request merged**